js/cobinhood.js

```diff
@@ -529,6 +529,82 @@ module.exports = class cobinhood extends Exchange {
         };
     }
 
+    async fetchDeposits (currency = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        if (typeof currency === 'undefined') {
+            throw new ExchangeError (this.id + ' needs currency for deposit history');
+        }
+        let curr = this.currency (currency);
+        let request = {
+            'limit': limit,
+            'currency': curr['id'],
+        };
+        let response = await this.privateGetWalletDeposits (this.extend (request, params));
+        return this.parseTransactions (response['result']['deposits'], 'deposit');
+    }
+
+    async fetchWithdrawals (currency = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        if (typeof currency === 'undefined') {
+            throw new ExchangeError (this.id + ' needs currency for deposit history');
+        }
+        let curr = this.currency (currency);
+        let request = {
+            'limit': limit,
+            'currency': curr['id'],
+        };
+        let response = await this.privateGetWalletWithdrawals (this.extend (request, params));
+        return this.parseTransactions (response['result']['withdrawals'], 'withdraw');
+    }
+
+    parseTransactions (transactions, side = undefined, market = undefined, since = undefined, limit = undefined) {
+        let result = Object.values (transactions || []).map (transaction => this.parseTransaction (transaction, side));
+        result = this.sortBy (result, 'timestamp');
+        let symbol = (typeof market !== 'undefined') ? market['symbol'] : undefined;
+        return this.filterBySymbolSinceLimit (result, symbol, since, limit);
+    }
+
+    parseTransactionStatus (status) {
+        if (['tx_pending_two_factor_auth', 'tx_pending_email_auth', 'tx_pending_approval', 'tx_approved', 'tx_processing', 'tx_pending', 'tx_sent'].indexOf (status) !== -1) {
```

Comment:
we can't use indexOf in a portable way for arrays (we can use it for strings only), therefore these checks just have to be written out one by one in if-else cases, or in the same way as done in other places – by having a map of exchange-specific statuses to exchange-agnostic (unified) statuses.

Reply:
Example: https://github.com/ccxt/ccxt/blob/master/js/binance.js#L560

Reply:
I see that parseTransactions method is not transpiled correctly to python. exchange.js parseTrades method is not really transpiled to python & php. So I suppose I must add parseTransactions method to both files (exchange.py, Exchange.php).